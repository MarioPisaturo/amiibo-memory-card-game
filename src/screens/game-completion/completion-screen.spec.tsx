import * as React from 'react';
import { shallow } from 'enzyme';

import CompletionScreen, { ICompletionScreen } from './completion-screen';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: 'localhost:3000/completion?moves=10',
  }),
}));

describe('CompletionScreen', () => {
  function makeProps(overrideProps: ICompletionScreen = {}): ICompletionScreen {
    return {
      ...overrideProps,
    };
  }

  it('should render a CompletionScreen', () => {
    const props: ICompletionScreen = makeProps();
    const wrapper = shallow(<CompletionScreen {...props} />);
    expect(wrapper).toHaveLength(1);
  });

  it('CompletionScreen should match snapshot', () => {
    const props: ICompletionScreen = makeProps();
    expect(shallow(<CompletionScreen {...props} />)).toMatchSnapshot();
  });
});

/** TODO: add other tests */
