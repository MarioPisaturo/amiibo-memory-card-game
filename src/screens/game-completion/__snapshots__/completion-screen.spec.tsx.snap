// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompletionScreen CompletionScreen should match snapshot 1`] = `ShallowWrapper {}`;
